### PR TITLE
More list initialization and NSDMI changes

### PR DESCRIPTION
**Pull request by DrItanium**
This is a continuation of my previous pull request. I have tended towards NSDMI in this pull request as it is more readable and easier to maintain as new constructors are added.

There is also a single use of std::unique_ptr in searchlineedit.cpp as it makes sense to use as a starting point. I am aware that Qt has a unique_ptr implementation which I believe is just an alias to std::unique_ptr these days. Over time, I want to replace raw pointers with unique and shared pointers as it makes memory management exponentially cleaner. We did the conversion over time at work and the program got smaller and easier to debug (this is a future plan so just view this single instance of what this will look like). 

Let me know what you think.

### CODE REVIEW

File: src/sketch/welcomeview.cpp [hunk 1]
```diff
@@ -930,22 +926,22 @@ void WelcomeView::getNextBlogImage(int ix, bool blog) {
...
+	auto blog = manager->property("blog").toBool();
 
-	int responseCode = networkReply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+	auto responseCode = networkReply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
 	if (responseCode == 200) {
 		QByteArray data(networkReply->readAll());
 		QPixmap pixmap;
 		if (pixmap.loadFromData(data)) {
-			QPixmap scaled = pixmap.scaled(QSize(ImageSpace, ImageSpace), Qt::KeepAspectRatio);
+			auto scaled = pixmap.scaled(QSize(ImageSpace, ImageSpace), Qt::KeepAspectRatio);
```
Comment (KjellMorgenstern) [inline src/sketch/welcomeview.cpp]: pixmap already is kind of Hungarian notation, auto doesn't improve this. QPixmap is a good guess for scaled() return type, but would you know it for pixmap.mask() ?

File: src/autoroute/autorouter.cpp [hunk 2]
```diff
@@ -274,14 +268,14 @@ void Autorouter::addUndoConnection(bool connect, TraceWire * traceWire, QUndoCom
...
-		        ViewLayer::specFromID(toConnectorItem->attachedToViewLayerID()),
-		        connect, parentCommand);
+		auto vw = qobject_cast<VirtualWire *>(toConnectorItem->attachedTo());
+		if (vw) continue;
+
+        auto ccc = new ChangeConnectionCommand(m_sketchWidget, crossView,
+                toConnectorItem->attachedToID(), toConnectorItem->connectorSharedID(),
+                connectorItem->attachedToID(), connectorItem->connectorSharedID(),
+                ViewLayer::specFromID(toConnectorItem->attachedToViewLayerID()),
+                connect, parentCommand);
```
[the remark this answers is not in the document]
Comment (DrItanium) [reply src/autoroute/autorouter.cpp]: I keep forgetting about using tabs... Fixed

**Comment by DrItanium** [comment]
I agree with not using auto in all cases as it can lead to problems deducing types. I tried to only choose ones which I knew what the type was. I went too far in some cases because I use YouCompleteMe in vim.

**Comment by KjellMorgenstern** [comment]
Thanks a lot for those cleanup!

**Pull request merged**